### ISSUE TITLE
failed to build lowlatency packages for v5.16.13

### ISSUE THREAD

**Issue by heuchi** [TuxInvader/focal-mainline-builder#16]
Thanks for providing this great tool. I've been using it to build my own lowlatency kernels for some weeks now. All worked fine until v5.16.11 (I didn't try v5.16.12). But with v5.16.13 it didn't work.
I invoked it with:
<pre>sudo docker run --cpus=7 -ti -e kver=v5.16.13 -v /usr/local/src/cod/mainline:/home/source -v /usr/local/src/cod/debs:/home/debs --rm tuxinvader/focal-mainline-builder:latest --flavour=lowlatency --exclude=cloud-tools,udebs</pre>
It ended like this and I got no packages in the debs folder:
<pre>
make[2]: Leaving directory '/home/source/debian/build/build-lowlatency'
make[1]: Leaving directory '/home/source'
Debug: /home/source/debian/stamps/stamp-prepare-indep
dh_prep -i
Debug: /home/source/debian/stamps/stamp-install-headers
dh_testdir
install -d /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613
find . -path './debian' -prune -o -path './debian.master' -prune \
  -o -path './include/*' -prune \
  -o -path './scripts/*' -prune -o -type f \
  \( -name 'Makefile*' -o -name 'Kconfig*' -o -name 'Kbuild*' -o \
     -name '*.sh' -o -name '*.pl' -o -name '*.lds' \) \
  -print | cpio -pd --preserve-modification-time /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613
23327 blocks
cp -a scripts include /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613
(find arch -name include -type d -print | \
        xargs -n1 -i: find : -type f) | \
        cpio -pd --preserve-modification-time /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/arm64/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/arm64/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/um/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/um/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/sh/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/sh/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/alpha/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/alpha/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/hexagon/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/hexagon/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/microblaze/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/microblaze/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/mips/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/mips/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/s390/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/s390/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/x86/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/x86/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/arm/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/arm/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/xtensa/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/xtensa/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/powerpc/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/powerpc/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/openrisc/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/openrisc/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/nios2/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/nios2/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/arc/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/arc/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/parisc/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/parisc/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/h8300/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/h8300/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/riscv/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/riscv/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/m68k/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/m68k/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/nds32/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/nds32/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/sparc/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/sparc/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/ia64/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/ia64/include/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/csky/include/uapi/asm/Kbuild not created: newer or same age version exists
cpio: /home/source/debian/linux-headers-5.16.13-051613/usr/src/linux-headers-5.16.13-051613/arch/csky/include/asm/Kbuild not created: newer or same age version exists
28722 blocks
+ grep -q CONFIG_MODULE_SIG=y /home/source/debian/build/build-lowlatency/.config
+ echo /home/source/debian/build/build-lowlatency/scripts/sign-file sha512 /home/source/debian/build/build-lowlatency/certs/signing_key.pem /home/source/debian/build/build-lowlatency/certs/signing_key.x509
Debug: /home/source/debian/stamps/stamp-install-lowlatency kernel_file arch/x86/boot/bzImage kernfile arch/x86/boot/bzImage install_file vmlinuz instfile vmlinuz
dh_testdir
dh_prep -plinux-image-unsigned-5.16.13-051613-lowlatency
dh_prep: error: Requested unknown package linux-image-unsigned-5.16.13-051613-lowlatency via -p/--package, expected one of: linux-source-5.16.13 linux-headers-5.16.13-051613 linux-tools-common linux-tools-5.16.13-051613 linux-cloud-tools-common linux-cloud-tools-5.16.13-051613 linux-tools-host linux-image-unsigned-5.16.13-051613-generic linux-modules-5.16.13-051613-generic linux-modules-extra-5.16.13-051613-generic linux-headers-5.16.13-051613-generic linux-image-unsigned-5.16.13-051613-generic-dbgsym linux-tools-5.16.13-051613-generic linux-cloud-tools-5.16.13-051613-generic linux-buildinfo-5.16.13-051613-generic linux-image-unsigned-5.16.13-051613-generic-64k linux-modules-5.16.13-051613-generic-64k linux-modules-extra-5.16.13-051613-generic-64k linux-headers-5.16.13-051613-generic-64k linux-image-unsigned-5.16.13-051613-generic-64k-dbgsym linux-tools-5.16.13-051613-generic-64k linux-cloud-tools-5.16.13-051613-generic-64k linux-buildinfo-5.16.13-051613-generic-64k linux-image-unsigned-5.16.13-051613-generic-lpae linux-modules-5.16.13-051613-generic-lpae linux-modules-extra-5.16.13-051613-generic-lpae linux-headers-5.16.13-051613-generic-lpae linux-image-unsigned-5.16.13-051613-generic-lpae-dbgsym linux-tools-5.16.13-051613-generic-lpae linux-cloud-tools-5.16.13-051613-generic-lpae linux-buildinfo-5.16.13-051613-generic-lpae
dh_prep: error: unknown option or error during option parsing; aborting
make: *** [debian/rules.d/2-binary-arch.mk:126: /home/source/debian/stamps/stamp-install-lowlatency] Error 255
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
********

Building meta package

********
********

Moving packages to debs folder

********
mv: cannot stat '/home/source/../*.*': No such file or directory
********
</pre>

**Comment by heuchi** [comment]
Since v5.16.14 came out pretty much the same moment I opened this issue, I also tried it, but got the same problem.

**Comment by TuxInvader** [comment]
Hi @heuchi 

It looks like the mainline kernel stopped providing `lowlatency` flavour kernels. It's hard to tell from the git log (there isn't one for the ubuntu parts), but it's gone AWOL.

```
$ git status
HEAD detached at cod/mainline/v5.16.12
nothing to commit, working tree clean
```

The control files for lowlatency on AMD64 and ARM64 have been removed.
```
$ git diff cod/mainline/v5.16.11 debian.master/control.d/
diff --git a/debian.master/control.d/vars.lowlatency b/debian.master/control.d/vars.lowlatency
deleted file mode 100644
index fe6994e29802..000000000000
--- a/debian.master/control.d/vars.lowlatency
+++ /dev/null
@@ -1,6 +0,0 @@
-arch="amd64 arm64"
-supported="Lowlatency"
-target="Geared toward desktop and server systems."
-desc="=HUMAN= SMP"
-bootloader="grub-pc [amd64] | grub-efi-amd64 [amd64] | grub-efi-ia32 [amd64] | grub [amd64] | lilo [amd64] | flash-kernel [armhf arm64] | grub-efi-arm64 [arm64] | grub-efi-arm [armhf]"
-provides="kvm-api-4, redhat-cluster-modules, ivtv-modules, virtualbox-guest-modules [amd64]"
diff --git a/debian.master/control.d/vars.lowlatency-64k b/debian.master/control.d/vars.lowlatency-64k
deleted file mode 100644
index 78617be9ebe4..000000000000
--- a/debian.master/control.d/vars.lowlatency-64k
+++ /dev/null
@@ -1,6 +0,0 @@
-arch="arm64"
-supported="Lowlatency 64K pages"
-target="Geared toward desktop and server systems."
-desc="=HUMAN= SMP"
-bootloader="grub-efi-arm64 [arm64] | flash-kernel [arm64]"
-provides="kvm-api-4, redhat-cluster-modules, ivtv-modules"
```

**Comment by heuchi** [comment]
Thanks for the explanation. So, I guess, I'll have to wait if/until they bring back the lowlatency support.

**Comment by TuxInvader** [comment]
I could incorporate the changes put together by Doug Smythes (https://askubuntu.com/users/372967/doug-smythies) here: https://askubuntu.com/questions/1397410/where-are-latest-mainline-low-latency-kernel-packages

**Comment by heuchi** [comment]
That would be great. I tried to use that config script myself (using --shell=yes option) but didn't succeed.

**Comment by TuxInvader** [comment]
I think I've recreated the lowlatency options with 76d83d10a25883d330a6bb180da4355fce03c731
Can you test please? It should be available on docker hub as image tuxinvader/focal-mainline-builder:v1.2

**Comment by heuchi** [comment]
This is great. I just compiled v5.17.1 without any problems and the settings (according to /boot/config-5.17.1-051701-lowlatency) match those given by Doug Smythes. Thanks.

**Comment by TuxInvader** [comment]
I tweaked the build script a little to be backwards compatible. Now it will only make the build config changes if you ask for a `lowlatency` build and the flavour is missing. On older kernels that have both flavours it will still build both if you don't specify. If the mainline team reintroduce lowlatency in the future, then we should automatically switch back to their config.

Pushed to docker as `tuxinvader/focal-mainline-builder:latest`
